### DISCUSSION TITLE
.csproj auto-detected as nx projects - how to ignore specific ones?

### DISCUSSION THREAD

**Discussion by Tungsten78** [nx-dotnet/nx-dotnet#688] [Q&A] [answered]
Use-case:   I have a generated library that is a .NET 4.5 swagger client (using @trumbitta/nx-plugin-openapi:generate-api-lib-sources executor).   It will be used externally in another repo, so I have no need to work with this within nx beyond the generation of the library.

Issue:  `nx affected` (build/lint/etc) auto-discovers the project and fails due to plugin requirements on .NET core.  specifically it's trying to restore .net packages.

What I've tried:
* Added path of projects to ignore to .nxignore - didn't make a difference
* --exclude <project name> - this works but I'd rather not litter this across all my cicd usages

Is there a way to force nx-dotnet to ignore specific .csproj files?

**Comment by AgentEnder** [comment, marked as answer]
You can use .nxignore if Nx shouldn't be concerned with that project at all

**Reply by Tungsten78** [reply]
Was hoping it would be this easy but affected still jumps into these. 

Of note, the detected projects are csproj files within an nx project. Does this have anything to do with it?  I have the root path of these in .nxignore

**Reply by AgentEnder** [reply]
What version of Nx are you on? Its possible that this is something we've recently adjusted. Also make sure the `.nxignore` file is in the workspace root, we don't read nested ones.

**Reply by Tungsten78** [reply]
Indeed `.nxignore` is at workspace root.

Playing further here, it seems to be picky about the path.  M

Folder structure
---
libs/generated/my-nx-codegen-project
libs/generated/my-nx-codegen-project/src/Org.OpenApiTools/

* project.json defined (only codegen target).  Runs executor to produce the following folder

.nxignore permutations
---
1. Without .nxignore
`nx print-affected --select=projects -> my-nx-codegen-project, org-openapitools`

2. With .nxignore having `libs/generated/my-nx-codegen-project/src`
`nx print-affected --select=projects -> my-nx-codegen-project, org-openapitools

3. With .nxignore having `libs/generated/my-nx-codegen-project`
`nx print-affected --select=projects -> ""

Option 2 was desired, I want to track the code-gen project, but not anything to be detected below it.
I will use option 3 as a work-around.

```
Nx Version:
- Local: v15.9.2
```

**Reply by AgentEnder** [reply]
You could also just disable the inference features from nx-dotnet, I think with having the `import-projects` generator and such there is minimal benefits from it. See: https://github.com/nx-dotnet/nx-dotnet/blob/master/packages/utils/src/lib/models/nx-dotnet-config.interface.ts#L25

**Comment by AgentEnder** [comment]
Update: In Nx 18, when specifying a plugin in nx.json you can set an exclude for things like this. It would look something like:

```json5
{
  "plugins": [{
    "plugin": "@nx-dotnet/core",
    "exclude": ["libs/generated"]
  }]
}
```
